.github/workflows/coverage_tests.yml

```diff
@@ -0,0 +1,146 @@
+          pip install -r .github/gh_requirements.txt
+
+      - name: Pull image from GHCR
+        run: docker pull ghcr.io/${{ github.repository }}:dev
+
+      # Only on scheduled runs, capture ISO weekday (1=Mon … 7=Sun).
+      - name: Set DAY_OF_WEEK
+        if: github.event_name == 'schedule'
+        run: echo "DAY_OF_WEEK=$(date -u +'%u')" >> $GITHUB_ENV
+
+      # This step is used to trigger the fast test coverage generation using the invoke task.
+      - name: Run Fast test and generate report
+        id: run_fast
+        continue-on-error: true
+        env:
+          GH_ACTION_ACCESS_TOKEN: ${{ secrets.GH_ACTION_ACCESS_TOKEN }}
+          CSFY_AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
+          CSFY_AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          CSFY_AWS_SESSION_TOKEN: ${{ env.AWS_SESSION_TOKEN }}
+          CSFY_AWS_DEFAULT_REGION: ${{ env.AWS_DEFAULT_REGION }}
+          CSFY_ECR_BASE_PATH: ghcr.io/${{ github.repository_owner }}
+          CSFY_AWS_S3_BUCKET: ${{ vars.CSFY_AWS_S3_BUCKET }}
+        run: invoke run_coverage --suite fast
+
+      - name: Upload Fast Test Coverage to Codecov
+        id: upload_fast
+        # Upload report if the fast test run successfully and report is generated.
+        if: steps.run_fast.outcome == 'success'
+        continue-on-error: true
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: ./coverage.xml
+          # Specify the Codecov flag name associated with this test suite.
+          # Required to separate coverage reports by type (e.g., fast, slow, superslow) inside the Codecov UI.
+          flags: fast
+          name: fast-test-coverage
+
+      - name: Run Slow test and generate report
+        id: run_slow
+        continue-on-error: true
+        env:
+          GH_ACTION_ACCESS_TOKEN: ${{ secrets.GH_ACTION_ACCESS_TOKEN }}
+          CSFY_AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
+          CSFY_AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          CSFY_AWS_SESSION_TOKEN: ${{ env.AWS_SESSION_TOKEN }}
+          CSFY_AWS_DEFAULT_REGION: ${{ env.AWS_DEFAULT_REGION }}
+          CSFY_ECR_BASE_PATH: ghcr.io/${{ github.repository_owner }}
+          CSFY_AWS_S3_BUCKET: ${{ vars.CSFY_AWS_S3_BUCKET }}
+        run: invoke run_coverage --suite slow
+
+      - name: Upload Slow Test Coverage to Codecov
+        id: upload_slow
+        if: steps.run_slow.outcome == 'success'
+        continue-on-error: true
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: ./coverage.xml
+          flags: slow
+          name: slow-test-coverage
+
+      - name: Run Superslow test and generate report
+        # Scheduled for Monday (i.e Day 1) or when manually triggered.
+        if: (github.event_name == 'schedule' && env.DAY_OF_WEEK == '1') || github.event_name == 'workflow_dispatch'
```

Comment:
I'd get rid of extra step "Set DAY_OF_WEEK" and put all the conditions inside the step. 
It could something like: 
```
run: |
         day_of_week=$(date +%u)
         if [ "$day_of_week" == "1" ] || [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
            invoke run_coverage --suite superslow
         else
            exit 1
         fi
```